test/hotspot/jtreg/compiler/vectorization/TestFloat16VectorConvChain.java

```diff
@@ -40,7 +40,9 @@
 public class TestFloat16VectorConvChain {
 
     @Test
-    @IR(applyIfCPUFeatureOr = {"f16c", "true", "avx512vl", "true", "zvfh", "true"}, counts = {IRNode.VECTOR_CAST_HF2F, IRNode.VECTOR_SIZE_ANY, ">= 1", IRNode.VECTOR_CAST_F2HF, IRNode.VECTOR_SIZE_ANY, " >= 1"})
+    @IR(applyIfCPUFeatureAnd = {"avx512_fp16", "false", "avx512vl", "true"}, counts = {IRNode.VECTOR_CAST_HF2F, IRNode.VECTOR_SIZE_ANY, ">= 1", IRNode.VECTOR_CAST_F2HF, IRNode.VECTOR_SIZE_ANY, " >= 1"})
+    @IR(applyIfCPUFeatureAnd = {"avx512_fp16", "false", "f16c", "true"}, counts = {IRNode.VECTOR_CAST_HF2F, IRNode.VECTOR_SIZE_ANY, ">= 1", IRNode.VECTOR_CAST_F2HF, IRNode.VECTOR_SIZE_ANY, " >= 1"})
```

Comment:
Wouldn't the Ideal transforms convert the IR for this test case to - 

ReinterpretS2HF     ReinterpretS2HF
                      \         /
                       AddHF
                            |
                       ReinterpretHF2S
                            |
                       ConvHF2F

Reply:
New transforms are guarded by target features checks, the IR test rules are enforced only on non AVX512_FP16 targets.